### DISCUSSION TITLE
Question on measurement units

### DISCUSSION THREAD

**Discussion by mdiepart** [Part-DB/Part-DB-server#589] [Q&A] [answered]
Hi,

I discovered this project a few days ago and I'm thinking about implementing it in our university to centralize electronic parts management. So I installed Part-DB on a machine and played with it a bit.

However, I'm having a hard time wrapping my head around the concept of measurement unit and the units for components values (which from hat I understood are different things). So it looks like what is called "Measurement Unit" is a way to measure how much of a specific item we have (i.e. pieces, boxes, meters, ...). What I don't understand is why a measurement unit can have parameters and what those parameters do. To test, I created a measurement unit named "Capacitor" and added a series of parameters that are commonly used for capacitors. But when I create a component for a capacitor and select "Capacitor" as a measurement unit, those parameters are nowhere to be seen, and I would have expected to find them pre-populated in the parameters list of the component.

So question 1 is, what are the parameters of a measurement unit used for? Can I pre-populate a list of parameters?

My second question is about the handling of SI unit prefixes. Especially when creating a parameter for the value of a component such as a capacitor. The capacitance value can often be expressed as "100nF" or "0.1uF". 

Question 2 would be: Is there a way to automatically handle the SI prefix units for parameters and, i.e. have a parameter named "Capacitance" with units "F" and a value which is either "100n" or "0.1u"?

Thank you for your help and for the nice software!

**Comment by jbtronics** [comment, marked as answer]
Measurement units are (currently) only for measuring parts amounts. By default parts without a special partUnit are assumed as just pieces. If you want to measure stock of something like a liquid, then you would define a measurement unit like liters, and assign it as part unit to the liquid "parts".
Parameters have their own unit field, which is basically just a text field, where you can input anything you want. Currently you have to ensure to input always the same unit, if you wanna compare parameters of different parts.

The parameters you can define on Measurement Units (like on all datastructures) are just to specify the defined unit further. These parameters are not inherited to parts. They only allow to specify additional informations about the entity in a structured way. For measurement units that maybe not so useful, but for something like footprints, you may want to put informations about the package sizes, etc. there.

There is currently no good way to prepopulate a list of parameters for a new parts. However if you already have some parameters defined, then you will get some autcomplete suggesting you this existing parameters, while inputting new ones. Also you can clone a part and modify the values, if you want to input a large batch of parts with similar parameters.

> Is there a way to automatically handle the SI prefix units for parameters and, i.e. have a parameter named "Capacitance" with units "F" and a value which is either "100n" or "0.1u"?

Currently this is not possible, but I think it should not be too difficult to implement. You should maybe ask an "feature request" issue for that.

**Comment by mdiepart** [comment]
Thank you for your detailed answer.

I will see if I can draft a pull request updating the documentation to make it more explicit about those measurement unit parameters.

I will also draft a feature request for units and values.

Have a nice day!
